### PR TITLE
Features/49

### PR DESCRIPTION
**Pull request by frantsao**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions
* Remember to set **idealista:develop** as base branch;

### Description of the Change
* Add MariaDB Debian packages support
* Fix file and dir permissions

### Benefits
* Easy install of MariaDB
* Improve security and reliability

### Possible Drawbacks
N/A

### Applicable Issues
#49

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -1,7 +1,24 @@
...
+mysql_deb_repo: "http://dev.mysql.com/get/{{ mysql_deb_package }}"
+
+mysql_required_libs:
+  - python-mysqldb
+
+mysql_packages:
+  - mysql-common
+  - mysql-server
+
+# Packages for MariaDB:
```
Comment (dortegau) [inline defaults/main.yml]: I would prefer two files for these configuration values depending on MySQL flavours. 

Something like `vars/mysql.yml` and `vars/mariadb.yml` setting that values.

File: tasks/install.yml [hunk 2]
```diff
@@ -2,24 +2,26 @@
...
-    dest: /tmp
+    dest: /usr/src
     validate_certs: "no"
+  when: mysql_flavour == 'mysql'
 
 - name: MySQL | Install MySQL repo deb
   apt:
-    deb: "/tmp/{{ mysql_deb_package }}"
+    deb: "/usr/src/{{ mysql_deb_package }}"
+  when: mysql_flavour == 'mysql'
```
Comment (dortegau) [inline tasks/install.yml]: I prefer diferent task files over zillions of `when` statements

File: defaults/main.yml (hunk 1 above)
Comment (frantsao) [reply defaults/main.yml]: Ok, changed.

File: tasks/install.yml (hunk 2 above)
Comment (frantsao) [reply tasks/install.yml]: Ok, changed.

File: .travis.yml [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
 sudo: required
 services:
   - docker
 install:
   - pip install pipenv
-  - pipenv install -r test-requirements.txt --python 2.7
+  - pipenv sync
 script:
-  - pipenv run molecule test
+  - pipenv run molecule test --scenario-name=mysql
```
Comment (jnogol) [inline .travis.yml]: why not `pipenv run molecule test --all`? This way, if any new scenario is added, there'll be no need of updating this file :)

File: templates/mariadb.service.j2 [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+# This file is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Thanks to:
+# Daniel Black
+# Erkan Yanar
+# David Strauss
+# and probably others
```
Comment (jnogol) [inline templates/mariadb.service.j2]: XD

File: README.md [hunk 5]
```diff
@@ -89,33 +89,34 @@ mysql_users:
...
+To check the installation, example with Oracle's MySQL implementation:
 
 ```bash
-$ pipenv run molecule converge
-$ pipenv run molecule login
+$ pipenv run molecule converge --scenario-name=mysql
+$ pipenv run molecule login --scenario-name=mysql
 
-vagrant@mysql:~$ mysql -u root -ptesting
+root@mariadb:/# mysql -u root -ptesting
```
Comment (jperera) [inline README.md]: "root@mariadb" ?

Comment (jnogol) [reply README.md]: guess is docker container's user

Comment (frantsao) [reply README.md]: fixed!

File: README.md [hunk 6]
```diff
@@ -24,7 +24,7 @@ These instructions will get you a copy of the role for your Ansible playbook. On
 
 ### Prerequisities
 
-Ansible 2.4.3.0 version installed.
+Ansible 2.7.9.0 version installed.
```
Comment (dortegau) [inline README.md]: I think that the possibility to install MariaDB with this role should be defined at the beginning in this README file

Comment (frantsao) [reply README.md]: Done!

**Pull request merged**